client/rpc/client.go

```diff
@@ -0,0 +1,164 @@
+package rpc
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"math/rand"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/nats-io/nats.go"
+	"github.com/rs/zerolog"
+	"github.com/statechannels/go-nitro/client/engine/store/safesync"
+	"github.com/statechannels/go-nitro/network"
+	"github.com/statechannels/go-nitro/network/serde"
+	natstrans "github.com/statechannels/go-nitro/network/transport/nats"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/protocols/directdefund"
+	"github.com/statechannels/go-nitro/protocols/directfund"
+	"github.com/statechannels/go-nitro/protocols/virtualfund"
+	"github.com/statechannels/go-nitro/types"
+
+	"github.com/statechannels/go-nitro/channel/state/outcome"
+)
+
+// RpcClient is a client for making nitro rpc calls
+type RpcClient struct {
+	nts       *network.NetworkService
+	myAddress types.Address
+	chainId   *big.Int
+
+	// responses is a collection of channels that are used to wait until a response is received from the RPC server
+	responses safesync.Map[chan interface{}]
+
+	idsToMethods safesync.Map[serde.RequestMethod]
+}
+
+// NewRpcClient creates a new RpcClient
+func NewRpcClient(rpcServerUrl string, myAddress types.Address, chainId *big.Int, logger zerolog.Logger) *RpcClient {
```

Comment:
This constructor implementation currently relies on NATS. Is it worth reflecting that in the name of the constructor? Or perhaps we can do that at a later stage when we add a client which works over a different (or over an abstract) transport. It looks like the underlying struct is built against an interface, so that's good.

Reply:
Good point. We should probably introduce a second transport to make sure our code is transport agnostic. This was done as a short term solution.

Filed https://github.com/statechannels/go-nitro/issues/1097.